File: Easy/1957. Delete Characters to Make Fancy String/sol.py
# https://leetcode.com/problems/delete-characters-to-make-fancy-string?envType=daily-question&envId=2024-11-01

class Solution:
    def makeFancyString(self, s: str) -> str:
        # Initialize the stack
        stack = []

        # Iterate over each character through the string
        for c in s:
            # We add current character to the stack
            stack.append(c)

            # If stack has more than 3 characters and and last three are same
            if len(stack) >=3 and (stack[-1] == stack[-2] == stack[-3]):
                # Remove the most recent character (preventing three consecutive same characters)
                stack.pop()
                
        # Join the stack to get the final string
        return "".join(stack)

File: Medium/1829. Maximum XOR for Each Query/sol.ts
// https://leetcode.com/problems/maximum-xor-for-each-query?envType=daily-question&envId=2024-11-08

function getMaximumXor(nums: number[], maximumBit: number): number[] {
    const mask: number = (1 << maximumBit) - 1;
    const n: number = nums.length;
    const res: number[] = new Array(n);
    let curr: number = 0;
    
    for(let i = 0; i < n; i++) {
        curr ^= nums[i];
        res[n-i-1] = (~curr & mask);
    }
    
    return res;
}